awx/main/dispatch/worker/base.py

```diff
@@ -132,22 +132,9 @@ def run(self, *args, **kwargs):
         super(AWXConsumerRedis, self).run(*args, **kwargs)
         self.worker.on_start()
 
-        time_to_sleep = 1
         while True:
-            while True:
-                try:
-                    res = self.redis.blpop(self.queues)
-                    time_to_sleep = 1
-                    res = json.loads(res[1])
-                    self.process_task(res)
```

Comment:
After this diff, `process_task` still exists and is being used, apparently by the postgres consumer. This method writes to the multiprocessing queue.

So let me echo my understanding - you are removing use of the multiprocessing queue _only for_ the redis-connected callback receiver. However, other messages, such as "delete this inventory" or "start running this job" are sent through the postgres messaging system, and still go through the dispatcher's multiprocessing queue. So one node in the cluster may have 4 or 8 redis connections, but 1 connection to the postgres message bus.

I'm just trying to state facts, and hopefully get the right. I don't recall any evidence that the main dispatcher ever had significant buildup of messages in the queue. Perhaps this is because autoscaling is enabled for the ordinary dispatcher but not for the callback receiver (going by my memory, it may be the other way around).

Reply:
> I don't recall any evidence that the main dispatcher ever had significant buildup of messages in the queue. Perhaps this is because autoscaling is enabled for the ordinary dispatcher but not for the callback receiver.

This is mostly because the volume is way lower - you might be running hundreds (or _maybe_ thousands) of parallel jobs/tasks, but it's not uncommon for the callback receiver to deal in far more events in parallel.

Reply:
The greatest volume we had there was the computed fields task. That did fail things in fun ways. I think those tended to be related to the autoscaling, memory, database connection limit, and so on. So filling up _that_ IPC queue didn't tend to come up much, and computed fields should be muzzled now anyway. So I agree with your assessment about volume.

Reply:
Yea, the dispatcher, unlike the callback receiver, can detect when all the workers are busy, and can autoscale up *new* workers to _avoid_ a backlog in the IPC queues.  There *is* a limit to this, of course, but on well-provisioned hardware and clusters, it's difficult to hit (we're talking hundreds to thousands of tasks).